agent/sshd/utmp.go

```diff
@@ -0,0 +1,229 @@
+package sshd
+
+/*	The Utmpx struct is derived from the Linux definition (man 5 utmpx).
+
+	At session start, a utmp record is constructed containing the PID,
+	tty name, ID, username, remote host, source IP and time, with
+	Type set to UserProcess. The record is written to UtmpxFile.
+	If a record with the same ID exists, that record is overwritten;
+	otherwise the record is appended to the file.  The same record is
+	appended to WtmpxFile.
+
+	At session end, Type is set to DeadProcess, the User and Host fields
+	are cleared, time is updated and the record is wrtten to UtmpxFile,
+	overwriting the session start record. The same record with ID and
+	source IP address cleared is appended to WtmpxFile.
+*/
+
+import (
+	"os"
+	"unsafe"
+	"bytes"
+	"encoding/binary"
+	"golang.org/x/sys/unix"
+	"github.com/sirupsen/logrus"
+)
+
+type ExitStatus struct {
+	ETermination	int16
+	EExit		int16
+}
+
+type Utmpx struct {
+	Type		int16		// UserProcess or DeadProcess
+	Pad		[2]byte	
```

Comment:
If `Pad` stands for padding, please use `Padding` as name here and also add the comment.

Reply:
Thanks for further comments.  I'll amend tomorrow.

Reply:
Done.  Revised PR to follow.